src/app/setup-wizard/pre-visit/company-setup/company-setup.component.ts

```diff
@@ -30,32 +31,45 @@ export class CompanySetupComponent implements OnInit, OnDestroy {
   energyUnit: FormControl;
   routeGuardWarningModal: boolean = false;
 
+  hasAssessments: boolean = false;
+  energyUnitChange: boolean = false;
+  filteredVisits: Array<IdbOnSiteVisit> = [];
+
   constructor(private router: Router,
     private companyIdbService: CompanyIdbService,
     private onSiteVisitIdbService: OnSiteVisitIdbService,
-    private companySetupService: CompanySetupService
+    private companySetupService: CompanySetupService,
+    private preAassessmentSetupService: PreAssessmentSetupService
   ) {
 
   }
 
   ngOnInit() {
     this.selectedCompanySub = this.companyIdbService.selectedCompany.subscribe(_company => {
       this.selectedCompany = _company;
-      // if (this.selectedCompany) {
-      //   this.companyName = _company.generalInformation.name;
-      // }
     });
     if (this.selectedCompany) {
       this.name = new FormControl(this.selectedCompany.generalInformation.name, [Validators.required]);
       this.energyUnit = new FormControl(this.selectedCompany.companyEnergyUnit, []); // Default value
       this.companySetupService.setControl(this.name);
+
+      this.filteredVisits = this.onSiteVisitIdbService.onSiteVisits?.getValue()
```

Comment:
Instead of going through the visit object.. Since we only care about the assessments here you could use the `assessmentIdbService` and add a `getByCompanyGuid` method. Then you'd have all the company assessments which is really all that's needed here.

Reply:
good idea, thanks